game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/pbem/PbemSetupPanel.java

```diff
@@ -0,0 +1,245 @@
+package games.strategy.engine.framework.startup.ui.posted.game.pbem;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.framework.message.PlayerListing;
+import games.strategy.engine.framework.startup.launcher.ILauncher;
+import games.strategy.engine.framework.startup.launcher.LocalLauncher;
+import games.strategy.engine.framework.startup.mc.GameSelectorModel;
+import games.strategy.engine.framework.startup.mc.HeadedLaunchAction;
+import games.strategy.engine.framework.startup.ui.PlayerSelectorRow;
+import games.strategy.engine.framework.startup.ui.PlayerType;
+import games.strategy.engine.framework.startup.ui.SetupPanel;
+import games.strategy.engine.framework.startup.ui.posted.game.DiceServerEditor;
+import games.strategy.engine.random.PbemDiceRoller;
+import java.awt.Dialog;
+import java.awt.Dimension;
+import java.awt.GridBagLayout;
+import java.awt.Window;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Optional;
+import javax.swing.Action;
+import javax.swing.JButton;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTabbedPane;
+import javax.swing.SwingUtilities;
+import javax.swing.border.EmptyBorder;
+import org.triplea.swing.SwingAction;
+import org.triplea.swing.jpanel.GridBagConstraintsAnchor;
+import org.triplea.swing.jpanel.GridBagConstraintsBuilder;
+import org.triplea.swing.jpanel.GridBagConstraintsFill;
+
+/**
+ * A panel for setting up Play by Email/Forum. This panel listens to the GameSelectionModel so it
+ * can refresh when a new game is selected or save game loaded The MainPanel also listens to this
+ * panel, and we notify it through the notifyObservers()
+ */
+public class PbemSetupPanel extends SetupPanel implements Observer {
+  private static final long serialVersionUID = 9006941131918034674L;
```

Comment:
This serialVersionUID is a duplicate of the other class, it should be changed so that it is unique, even though we don't really make use of this property.

Reply:
Could be nice to make it unique, but not required: https://github.com/triplea-game/triplea/pull/5347#discussion_r334496182

Reply:
```suggestion
  private static final long serialVersionUID = -4027051961383144244L;
```

Reply:
I don't think anyone would know this value is duplicated anywhere, except us for now.. If feeling important for the serialversionuid to be different, then the 0L and 1L values should arguably be updated too. I don't think there is a favorable ROI for even really worrying about this too much.